isushibsiteaccesssync.module

```diff
@@ -0,0 +1,218 @@
+<?php
+
+/**
+ * @file
+ * Syncs the access list with Active Directory group(s).
+ */
+
+
+/**
+ * Implements hook_menu().
+ */
+function isushibsiteaccesssync_menu() {
+  $items['admin/config/people/isushibsiteaccesssync'] = array(
+    'title' => 'Configure AD Sync',
+    'description' => 'Edit settings for Active Directory sync.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('isushibsiteaccesssync_settings'),
+    'access arguments' => array('administer shibboleth site access ad sync'),
+    'file' => 'isushibsiteaccesssync.admin.inc',
+  );
+
+  return $items;
+}
+
+
+/**
+ * Implements hook_cron().
+ */
+function isushibsiteaccesssync_cron() {
+  // Only run this once per day
+  if (variable_get('isushibsiteaccesssync_day_of_year', 0) == date('z') + 1) {
```

Comment:
Either way should work. The why it's written, on Jan 1, isushibsiteaccesssync_day_of_year = 365 and date('z') + 1 should equal 1. I won't be equal so the if statement will fail, and the rest of the function will run. At the end of the function, the function will be set to 1.

Your suggestion does have the advantage of being able to easily knowing the date it last ran.